### PR TITLE
[DOCS] EQL: Document `size` limit for pipes

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:
* Documents the `size` default as `10`.
* Updates `size` param def to note its relation to pipes.
* Updates the `head` and `tail` pipe docs to modify sequences.
* Documents the `fetch_size` parameter.

Relates to #59014 and #59063

### CODE REVIEW

**Comment by jrodewig** [comment]
Converting this to a draft so I can include changes from #59063

**Comment by rw-access** [comment]
@costin quick question on the default size. How should we make EQL in Elasticsearch and Python consistent? The python default when unspecified has been 50. Should that existing behavior be changed to default instead to 10? Or should this be changed to match the existing default of 50?

File: docs/reference/eql/eql-search-api.asciidoc [hunk 1]
```diff
@@ -160,6 +160,18 @@ Defaults to `event.category`, as defined in the {ecs-ref}/ecs-event.html[Elastic
 Common Schema (ECS)]. If an index does not contain the `event.category` field,
 this value is required.
 
+`fetch_size`::
```
Comment (costin) [inline docs/reference/eql/eql-search-api.asciidoc:163]: Currently, this value has to be larger than 2. Note sure though whether it makes sense mentioning this.

**Comment by costin** [comment]
> @costin quick question on the default size. How should we make EQL in Elasticsearch and Python consistent? The python default when unspecified has been 50. Should that existing behavior be changed to default instead to 10? Or should this be changed to match the existing default of 50?

Let's discuss this in our weekly meeting. In Elasticsearch, one would be able specify the default size (https://github.com/elastic/elasticsearch/issues/58646) and default ordering.

Value `10` was used since this is the default for Elasticsearch document search.

> Should that existing behavior be changed to default instead to 10?

That would be my preference in order to make things consistently across different Elasticsearch products.

File: docs/reference/eql/eql-search-api.asciidoc (hunk 1 above)
Comment (costin) [reply docs/reference/eql/eql-search-api.asciidoc:163]: I would add that this value cannot be higher `index_max_result_window` (one can try but will end up with an exception from ES).

**Review by costin — COMMENTED** [review]
LGTM

**Review by costin — APPROVED** [review]

**Pull request merged**